File: compiler/src/main/java/compiler/instruction/ClassInstruction.java
package compiler.instruction;

/**
 *
 * @author sasza
 */
public class ClassInstruction implements Instruction {
    
    private final String name;
    
    public ClassInstruction(String name){
        this.name = name;
    }

    @Override
    public String toAssembly() {
        return "CLASS " + name;
    }
    
}

File: virtual_machine/vm/instruction/types/ClassInstruction.cpp
#include "ClassInstruction.h"

Instruction * ClassInstruction::fromList(std::vector <std::string> mnemonics){    
    ClassInstruction * instruction = new ClassInstruction();
    instruction->setName(mnemonics[1]);    
    return instruction;
}

std::string ClassInstruction::getName(){
    return this->name;
}

void ClassInstruction::setName(std::string name){
    this->name = name;
}

void ClassInstruction::execute(){
    
}

std::string ClassInstruction::toString(){
    return "CLASS " + name;
}